src/control/cmd/daos/pool.go

```diff
@@ -337,51 +337,61 @@ func queryPoolHelper(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*da
 
 	return poolInfo, nil
 }
-
 func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
-	origQueryMask := queryMask
-	queryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines,
-		daos.PoolQueryOptionSuspectEngines, daos.PoolQueryOptionDisabledEngines)
-	if origQueryMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
-		queryMask.SetOptions(daos.PoolQueryOptionEnabledEngines)
-		origQueryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines)
-	} else if origQueryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
-		queryMask.SetOptions(daos.PoolQueryOptionDisabledEngines)
-		origQueryMask.ClearOptions(daos.PoolQueryOptionDisabledEngines)
-	} else if origQueryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
-		queryMask.SetOptions(daos.PoolQueryOptionSuspectEngines)
-		origQueryMask.ClearOptions(daos.PoolQueryOptionSuspectEngines)
-	}
-	poolInfo, err := queryPoolHelper(poolHdl, queryMask)
-	if err != nil {
-		return nil, err
-	}
-	if origQueryMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
+	poolInfo := &daos.PoolInfo{}
+	originalMask := queryMask // Save the original queryMask
+
+	// Function to handle the query and return a single RankList
```

Comment:
nit: In order to avoid unintended consequences, in-line functions like this should accept the object they are acting on as a parameter rather than mutating objects in an outer scope.

Reply:
This is a good point. I think in this case it is safe, but it is easier to reason about if it accepts the object to be acted upon.